README.md

```diff
@@ -1,2 +1,58 @@
 # TargetedTilingPrimers
 Create mutation targeting primers tiling codons of a gene for codon mutagenesis
+
+This repository contains information and a Python script for designing primers to generate mutant libraries of genes with specific desired mutations at each site.
+
+## Codon mutagenesis
+
+Codon mutagenesis can be used to create mutant libraries of a gene with all possible codon mutations.
+Such libraries are useful for experiments such as [deep mutational scanning](https://www.ncbi.nlm.nih.gov/pubmed/25075907) or [mutational antigenic profiling](http://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1006271)
+
+A [script in a separate repository](https://github.com/jbloomlab/CodonTilingPrimers) can be used to generate random mutations at each codon in a gene. This protocol randomly mutates each codon to all other possible codons. This protocol and papers using it are also described in that repository.
+
+Rather than generating mutant libraries with all possible codon mutations at every site in a gene, the script in this repository can be used to generate primers that will result in specific desired mutations at each site in a gene. In this way, a mutant library can be generated which has fewer lethal mutations than random codon mutagenesis.
+
+## Running the script to design primers
+
+The [create_primers.py](create_primers.py) Python script can be used to create primers that tile the codons of a gene in both the forward and reverse direction. Each pair of foward and reverse primers will produce one specified mutation at one specified site. The desired mutations must be given as a csv file argument.
+
+### Required arguments
+
+*seq* : Text file containing the sequence of the gene. The gene itself should be upper case, including the start and stop codons. Any flanking regions should be lower case. There must be >= (primerlength - 3) / 2.0 nucleotides before the first codon mutagenized and after the last codon mutagenized, otherwise there would not be enough flanking nucleotides for a primer to mutate those codons.
+
+*mutations_csv* : csv file containing a table with the mutations the primers will make to seq. The table should have a column, 'site', with site numbers, and a column, 'mutant', with the single letter amino acid mutant to make at that site. Each site can have multiple mutants, or none. The site number should denote the number of the codon in the uppercase gene, with the start codon being 1.
```

Comment:
I chose to have each mutation be a line in the file so that it would be possible later to add the ability to specify how many codons to generate for that mutation or how to choose the codons (which would take adding another column).

Reply:
Sounds good, you should just make it clear then that if a site has multiple mutants, it needs to be listed separately for each mutant. That's not clear from the description.